test/ts/jsx-namespacce-test.tsx

```diff
@@ -0,0 +1,16 @@
+import { createElement, Component } from "../../src";
+
+// declare global JSX types that should not be mixed with preact's internal types
+declare global {
+	namespace JSX {
+		interface Element {
+			unknownProperty: string
+		}
+	}
+}
```

Comment:
This snippet simulates what `@types/react` does. Will you be interested in having the real React typings as a dev-dependency instead of this mock?